pkg/actions/addon/tasks.go

```diff
@@ -119,15 +119,15 @@ type deleteAddonIAMTask struct {
 func (t *deleteAddonIAMTask) Describe() string { return t.info }
 
 func (t *deleteAddonIAMTask) Do(errorCh chan error) error {
-	defer close(errorCh)
-
 	errMsg := fmt.Sprintf("deleting addon IAM %q", *t.stack.StackName)
 	if t.wait {
 		if err := t.stackManager.DeleteStackBySpecSync(t.ctx, t.stack, errorCh); err != nil {
 			return fmt.Errorf("%s: %w", errMsg, err)
 		}
 		return nil
 	}
+
+	defer close(errorCh)
```

Comment:
Closing error channel to early here was nullifying the effect of the `--wait` flag

Reply:
It actually introduced a bug - https://github.com/eksctl-io/eksctl/issues/7177. Going to address it in a separate PR.